displayChallenges.js

```diff
@@ -0,0 +1,282 @@
+            }
+        })
+    }
+    for (let i = 0; i < topThreeByRating.length; i++) {
+        changeCardContent(topThreeInIndex[i], topThreeByRating[i]);
+    }
+}
+
+
+function changeCardContent(cardToBeChanged, theNewCard) {
+    cardToBeChanged.querySelector("img").src = theNewCard.image;
+    cardToBeChanged.querySelector("h3").innerHTML = theNewCard.title;
+    const stars = cardToBeChanged.querySelectorAll("ul > li");
+    for (let i = 0; i < stars.length; i++) {
+        if (i < theNewCard.rating) {
+            stars[i].className = "rating-star active";
+        } else {
+            stars[i].className = "rating-star";
+        }
+    }
+    const participants = cardToBeChanged.querySelector("small");
+    participants.innerHTML = theNewCard.minParticipants + "-" + theNewCard.maxParticipants + " participants";
+    participants.dataset.minParticipants = theNewCard.minParticipants;
+    participants.dataset.maxParticipants = theNewCard.maxParticipants;
+    cardToBeChanged.querySelector("p").innerHTML = theNewCard.description;
+    const button = cardToBeChanged.querySelector("button");
+    if (theNewCard.type == "online") {
+        button.innerHTML = "Take challenge online";
+    } else {
+        button.innerHTML = "Book this room";
+        button.className = "button secondary";
+    }
+}
+
+//makes the screen change upon resizing.
+window.onresize = doIt;
+
+// This function resizes the screen
+
+function doIt() {
+    if (document.location == "http://127.0.0.1:5500/challenges.html") {
+        const listElements = document.querySelectorAll("section > ul > li");
+        if (window.innerWidth < 1000) {
+            for (const element of listElements) {
+                element.style.width = "auto";
+                element.style.margin = "auto";
+                element.style.alignContent = "center";
+            }
+        } else {
+            for (const element of listElements) {
+                element.style.width = "calc(95%/3)";
+                element.style.maxWidth = "400px";
+                element.style.margin = "0";
+                challengesHtmlList.lastChild.style.marginRight = "auto";
+                challengesHtmlList.lastChild.style.paddingLeft = "2.5%";
+            }
+        }
+    }
+}
+
+
+// Buttons for online challenges
+
+const onlineButtons = document.querySelectorAll("a.button.primary");
+onlineButtons.forEach(button => {
+    button.addEventListener("click", (ev) => {
+        document.cookie = "online";
+        document.location.href = "http://127.0.0.1:5500/challenges.html";
+    })
+})
+
+
+// Buttons for on-site challenges
+
+const onSiteButtons = document.querySelectorAll("a.button.secondary");
+onSiteButtons.forEach(button => {
+    button.addEventListener("click", (ev) => {
+        document.cookie = "offline";
+        document.location.href = "http://127.0.0.1:5500/challenges.html";
+    })
+})
+
+// THE function that interwines everything
+
+function wellHello() {
```

Comment:
This is a bad function name, because it says nothing about what this function does. I know it can be annoying having to pause and come up with a good name when you're eager to write the actual code for what the function _does_, but if you don't like pausing beforehand, find a way to remember coming back to it later, before you check code into git.